### PR TITLE
Authenticate the api/token endpoint using a canary

### PR DESCRIPTION
**Pull request by GuillaumeSimonnet**

### CODE REVIEW

File: samples/advanced-csharp/Controllers/ApiController.cs [hunk 1]
```diff
@@ -31,6 +32,24 @@ public ApiController(Microsoft.Extensions.Configuration.IConfiguration configura
         [Route("token")]
```
Comment (metanMSFT) [inline samples/advanced-csharp/Controllers/ApiController.cs]: Need to add [HttpPost] here?

File: samples/advanced-csharp/Canary.cs [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+using System.Security.Cryptography;
+using System.Text;
+
+namespace AdvancedSampleWebApp
+{
+    public class Canary
+    {
+        public const int KeyLength = 32;
+        public const int NonceLength = 32;
+        public static TimeSpan ExpirationSpan = TimeSpan.FromHours(24);
```
Comment (metanMSFT) [inline samples/advanced-csharp/Canary.cs]: Readonly

File: samples/advanced-csharp/Canary.cs [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+        private static byte[] _key;
```
Comment (metanMSFT) [inline samples/advanced-csharp/Canary.cs]: Readonly

File: samples/advanced-csharp/Controllers/ApiController.cs (hunk 1 above)
Comment (GuillaumeSimonnet) [reply samples/advanced-csharp/Controllers/ApiController.cs]: Added, as well as content types supported.

File: samples/advanced-csharp/Canary.cs (hunk 2 above)
Comment (GuillaumeSimonnet) [reply samples/advanced-csharp/Canary.cs]: Arguably this could be a runtime parameter.

**Pull request merged**